models/repo.go

```diff
@@ -2240,6 +2240,14 @@ func GitFsck() {
 			func(idx int, bean interface{}) error {
 				repo := bean.(*Repository)
 				repoPath := repo.RepoPath()
+				// TODO: Move this elsewhere
+				if gitRepo, err := git.OpenRepository(repoPath); err == nil {
+					log.Trace("Building commit graph index")
+					if err := gitRepo.BuildCommitGraph(false); err != nil {
+						desc := fmt.Sprintf("Failed to build commit graph (%s): %v", repoPath, err)
+						log.Warn(desc)
+					}
+				}
```

Comment:
**This is inserted here only to allow people to play with the feature.**

Running a health check for all the repositories will also rebuild the commit graph files (http://gitea/admin?op=9). It is entirely possible to generate the commit graph file using the command line `git commit-graph write` tool instead. The bloom filter experiment is enabled by changing `BuildCommitGraph(false)` to `BuildCommitGraph(true)` in the above code. It will significantly increase the size of the commit graph files and the time to build it, but in many cases it will also significantly speed up hhistory queries on large repositories (unless I broke it :D).